kroxylicious-filters/kroxylicious-record-encryption/src/test/java/io/kroxylicious/filter/encryption/RecordEncryptionFilterTest.java

```diff
@@ -221,6 +225,28 @@ void shouldPropagateRequestExceptions() {
         assertThat(stage).failsWithin(Duration.ZERO).withThrowableThat().isInstanceOf(ExecutionException.class).havingCause().isSameAs(exception);
     }
 
+    @Test
+    void shouldRespondWithErrorOnRequestNotSatisfiableExceptions() {
+        // Given
+        final RequestFilterResultBuilder resultBuilder = stubErrorResponse();
+
+        var produceRequestData = buildProduceRequestData(new TopicProduceData()
+                .setName(ENCRYPTED_TOPIC)
+                .setPartitionData(List.of(new PartitionProduceData().setRecords(makeRecord(HELLO_CIPHER_WORLD)))));
+        final RequestNotSatisfiable failure = new RequestNotSatisfiable("could not acquire a DEK");
+        when(kekSelector.selectKek(anySet())).thenReturn(CompletableFuture.failedFuture(failure));
+
+        // When
+        CompletionStage<RequestFilterResult> stage = encryptionFilter.onProduceRequest(ProduceRequestData.HIGHEST_SUPPORTED_VERSION,
+                new RequestHeaderData(), produceRequestData, context);
+
+        // Then
+        assertThat(stage)
+                .succeedsWithin(Duration.ofSeconds(10));
+        verify(context).requestFilterResultBuilder();
+        verify(resultBuilder).errorResponse(any(), any(), argThat(throwable -> assertThat(throwable.getClass()).isAssignableTo(ApiException.class)));
```

Comment:
Arguably this should be tightened to `NetworkException` but I think `ApiException` is probalby right